### ISSUE TITLE
[Wayland] [kwin_wayland] sddm-helper-start-wayland fails/exits after manually turn on monitors

### ISSUE THREAD

**Issue by jfroy** [sddm/sddm#1847]
# Summary

I turned off my 2 monitors overnight with the computer at SDDM waiting for login and turned them back on in the morning. This led to `sddm-helper-start-wayland` to fail and SDDM to exit, leaving me with a broken/empty TTY.

Logs:

```
Jan 03 06:53:58 wave5 sddm-helper-start-wayland[44348]: "kwin_core: Parse error in tiles configuration for monitor \"92e842d7-5928-5c43-884a-4912e7cc82ed\" : \"illegal value\" Creating default setup\n"
Jan 03 06:53:58 wave5 sddm-greeter[44353]: Creating a fake screen in order for Qt not to crash
Jan 03 06:53:58 wave5 sddm-greeter[44353]: Loading file:///usr/share/sddm/themes/breeze/Main.qml...
Jan 03 06:53:58 wave5 sddm-greeter[44353]: Failed to find a Kirigami platform plugin
Jan 03 06:53:58 wave5 sddm-greeter[44353]: Adding view for "" QRect(0,0 0x0)
Jan 03 06:53:59 wave5 sddm-helper-start-wayland[44348]: "kwin_core: Could not find any suitable output for a layer surface\n"
Jan 03 06:53:59 wave5 sddm-helper-start-wayland[44348]: wayland greeter finished 0 QProcess::NormalExit
Jan 03 06:53:59 wave5 sddm-helper-start-wayland[44348]: quitting helper-start-wayland
Jan 03 06:53:59 wave5 sddm-helper-start-wayland[44348]: Stopping... "kwin_wayland"
Jan 03 06:53:59 wave5 sddm-helper-start-wayland[44348]: "QtDBus: cannot relay signals from parent QObject(0x55ad50323b20 \"\") unless they are emitted in the object's thread QThread(0x55ad50183238 \"libinput-connection\"). Current thread is QThread(0x55ad501328f0 \"\").\n"
Jan 03 06:53:59 wave5 xdg-desktop-por[44492]: Error reading events from display: Broken pipe
Jan 03 06:53:59 wave5 systemd[44295]: xdg-desktop-portal-gtk.service: Main process exited, code=exited, status=1/FAILURE
Jan 03 06:53:59 wave5 systemd[44295]: xdg-desktop-portal-gtk.service: Failed with result 'exit-code'.
Jan 03 06:53:59 wave5 sddm-helper-start-wayland[44348]: wayland compositor finished 15 QProcess::NormalExit
Jan 03 06:53:59 wave5 sddm-helper-start-wayland[44348]: quitting helper-start-wayland
Jan 03 06:53:59 wave5 sddm-helper[44279]: [PAM] Closing session
Jan 03 06:53:59 wave5 sddm-helper[44279]: pam_unix(sddm-greeter:session): session closed for user sddm
Jan 03 06:53:59 wave5 sddm-helper[44279]: [PAM] Ended.
Jan 03 06:53:59 wave5 sddm[1193]: Auth: sddm-helper exited successfully
Jan 03 06:53:59 wave5 systemd-logind[1050]: Session c6 logged out. Waiting for processes to exit.
Jan 03 06:53:59 wave5 sddm[1193]: Greeter stopped. SDDM::Auth::HELPER_SUCCESS
Jan 03 06:53:59 wave5 systemd[1]: session-c6.scope: Deactivated successfully.
```

I had to open a console session in another TTY and restart `sddm.service`.

# Configuration

/etc/sddm.conf.d/10-wayland.conf 
```
[General]
DisplayServer=wayland
GreeterEnvironment=QT_WAYLAND_SHELL_INTEGRATION=layer-shell

[Wayland]
CompositorCommand=kwin_wayland --drm --no-lockscreen --no-global-shortcuts --locale1
```

/etc/sddm.conf.d/kde_settings.conf
``` 
[Autologin]
Relogin=false
Session=
User=

[General]
HaltCommand=/usr/bin/systemctl poweroff
RebootCommand=/usr/bin/systemctl reboot

[Theme]
Current=breeze
CursorSize=
CursorTheme=breeze_cursors
Font=Noto Sans,10,-1,0,400,0,0,0,0,0,0,0,0,0,0,1

[Users]
MaximumUid=60513
MinimumUid=1000
```

# System info

Arch Linux
Linux: 6.6.9-arch1-1
Nvidia: 545.29.06 (open dkms)
sddm: 0.20.0-3

**Comment by jfroy** [comment]
See also kwin bug https://bugs.kde.org/show_bug.cgi?id=479359.

**Comment by HighwayStar** [comment]
I have same issue on Fedora 40

**Comment by Catoverflow** [comment]
I have the same issue on Arch Linux. I start my desktop with monitor off, and then turn it back on, resulting in a broken tty. logs:

```
Aug 06 17:55:02 Theseus sddm-greeter-qt6[687]: Creating a fake screen in order for Qt not to crash
Aug 06 17:55:02 Theseus sddm-greeter-qt6[687]: Loading file:///usr/share/sddm/themes/breeze/Main.qml...
Aug 06 17:55:02 Theseus sddm-greeter-qt6[687]: Adding view for "" QRect(0,0 0x0)
Aug 06 17:55:02 Theseus sddm-helper-start-wayland[671]: "kwin_core: Could not find any suitable output for a layer surface\n"
Aug 06 17:55:02 Theseus sddm-helper-start-wayland[671]: wayland greeter finished 0 QProcess::NormalExit
Aug 06 17:55:02 Theseus sddm-helper-start-wayland[671]: quitting helper-start-wayland
Aug 06 17:55:02 Theseus sddm-helper-start-wayland[671]: Stopping... "kwin_wayland"
Aug 06 17:55:02 Theseus sddm-helper-start-wayland[671]: wayland compositor finished 15 QProcess::NormalExit
Aug 06 17:55:02 Theseus sddm-helper-start-wayland[671]: quitting helper-start-wayland
Aug 06 17:55:02 Theseus sddm-helper[656]: [PAM] Closing session
Aug 06 17:55:02 Theseus sddm-helper[656]: pam_unix(sddm-greeter:session): session closed for user sddm
Aug 06 17:55:02 Theseus sddm-helper[656]: pam_systemd(sddm-greeter:session): New sd-bus connection (system-bus-pam-systemd-656) opened.
Aug 06 17:55:02 Theseus sddm-helper[656]: [PAM] Ended.
Aug 06 17:55:02 Theseus sddm[653]: Auth: sddm-helper exited successfully
Aug 06 17:55:02 Theseus sddm[653]: Greeter stopped. SDDM::Auth::HELPER_SUCCESS
Aug 06 17:55:02 Theseus systemd-logind[616]: Session c1 logged out. Waiting for processes to exit.
Aug 06 17:55:02 Theseus systemd[1]: session-c1.scope: Deactivated successfully.
Aug 06 17:55:02 Theseus systemd[1]: session-c1.scope: Consumed 1.324s CPU time, 376.9M memory peak.
Aug 06 17:55:02 Theseus systemd-logind[616]: Removed session c1.
```
My sddm config under `/etc/sddm.conf.d` is the same as op's

**Comment by Vogtinator** [comment]
That's a kwin issue, please report that on bugs.kde.org.

**Comment by dnlbtz** [comment]
This still happens and it looks like an sddm bug rather than a kwin one: in the
log above the greeter exits on its own with status 0 and kwin is only stopped
afterwards.

GreeterApp destroys its QQuickView when the QScreen goes away and
quitOnLastWindowClosed is never turned off, so with a single display the
greeter quits as soon as the monitor is gone. Reproduced with the Qt 6 greeter
on develop, so it is not specific to the Qt 5 one.

Filed as #2211, fix in #2212.
